homeassistant/components/rainmachine/__init__.py

```diff
@@ -158,6 +160,29 @@ async def async_update(api_category: str) -> dict:
 
     entry.async_on_unload(entry.add_update_listener(async_reload_entry))
 
+    async def async_pause_watering(call: ServiceCall) -> None:
+        """Pause watering for a set number of seconds."""
+        LOGGER.error(call.data)
+        await controller.watering.pause_all(call.data[CONF_SECONDS])
+        await async_update_programs_and_zones(hass, entry)
+
+    async def async_stop_all(_: ServiceCall) -> None:
+        """Stop all watering."""
+        await controller.watering.stop_all()
+        await async_update_programs_and_zones(hass, entry)
+
+    async def async_unpause_watering(_: ServiceCall) -> None:
+        """Unpause watering."""
+        await controller.watering.unpause_all()
+        await async_update_programs_and_zones(hass, entry)
+
+    for service_name, method in (
```

Comment:
Only register the services when setting up the first config entry. We can check if one service is registered with `hass.services.has_service`.

Then when unloading the config entry, remove the services if it's the last config entry remaining.